### PR TITLE
Add RedshiftDialect_redshift_connector

### PR DESCRIPTION
**Pull request by Brooke-white**
Adds a dialect to support use of redshift_connector database driver.

Modifies sqlalchemy-redshift __init__ requirement for psycopg2 to dialect `dbapi` method to avoid raising an ImportError for users using RedshiftDialect_redshift_connector

Modifies pytest to accept a command line option, `dbdriver`, to provide 1 or more database drivers to run the test suite against (e.g. `—dbdriver psycopg2 —dbdriver redshift_connector`). By default tests will run for `psycopg2` and `psycopg2cffi`

Modifies tox configuration to test with `—dbdriver redshift_connector` for compatible Python versions (i.e. py36 >=), and to use a version of `requests`. 

Reworks `test_dialects.py` to use a mock sqlalchemy engine.

## Todos
- [x] MIT compatible
- [x] Tests
- [x] Documentation
- [x] Updated CHANGES.rst

### CODE REVIEW

**Comment by Brooke-white** [comment]
These changes to the tests seem to be slowing down things quite a bit, I'll take another look on Monday.

File: sqlalchemy_redshift/dialect.py [hunk 1]
```diff
@@ -900,7 +901,16 @@ def create_connect_args(self, *args, **kwargs):
 class RedshiftDialect_psycopg2(
     Psycopg2RedshiftDialectMixin, PGDialect_psycopg2
 ):
-    pass
+    @classmethod
+    def dbapi(cls):
```
Comment (jklukas) [inline sqlalchemy_redshift/dialect.py:905]: Can this be defined in `Psycopg2RedshiftDialectMixin` or does it have to be a classmethod on each concrete dialect? I assume this is some hook in the SQLAlchemy machinery, but I'm struggling to find documentation about a dbapi classmethod. Can you point to background on this approach?

Comment (Brooke-white) [reply sqlalchemy_redshift/dialect.py:905]: I think we could move this to `Psycopg2RedshiftDialectMixin` if we did something like this, using the value of `driver` defined by the dialect (e.g. defined [here](https://github.com/sqlalchemy/sqlalchemy/blob/fb81f9c8d914f9911925dd3f4e77d7fc374b267c/lib/sqlalchemy/dialects/postgresql/psycopg2cffi.py#L29) for the psycopg2cfifi dialect).

```python
class Psycopg2RedshiftDialectMixin(RedshiftDialectMixin): 
   @classmethod
    def dbapi(cls):
        try:
            return __import__(self.driver)
        except ImportError:
            raise ImportError(
                'No module named {}.'.format(self.driver)
            )
```

For the `dbapi` classmethod, looking to the `DefaultDialect` class (from which [`PGDialect`](https://github.com/sqlalchemy/sqlalchemy/blob/fb81f9c8d914f9911925dd3f4e77d7fc374b267c/lib/sqlalchemy/dialects/postgresql/base.py#L3076) inherits from) it is used in [`DefaultDialect.connect()`](https://github.com/sqlalchemy/sqlalchemy/blob/bf1fe670513abeb1596bc5266f50db1ffe62f3bd/lib/sqlalchemy/engine/default.py#L582), as well as for defining the [dbapi paramstyle that will be used](https://github.com/sqlalchemy/sqlalchemy/blob/bf1fe670513abeb1596bc5266f50db1ffe62f3bd/lib/sqlalchemy/engine/default.py#L307).

I was unable to find any docs for the `dpapi` classmethod either, as I stumbled upon it when looking for where the driver's `connect` method is called.

File: sqlalchemy_redshift/dialect.py [hunk 2]
```diff
@@ -896,6 +897,15 @@ def create_connect_args(self, *args, **kwargs):
         default_args.update(cparams)
         return cargs, default_args
 
+    @classmethod
+    def dbapi(cls):
+        try:
+            return __import__(cls.driver)
```
Comment (jklukas) [inline sqlalchemy_redshift/dialect.py:903]: It looks like https://docs.python.org/3/library/importlib.html#importlib.import_module is preferred over `__import__`.

**Review by jklukas — COMMENTED** [review]
This looks good. I've kicked off integration tests. Do you have any more info on duration of tests with this change?

**Comment by Brooke-white** [comment]
> Do you have any more info on duration of tests with this change?

It seems like the tests which make a connection to a Redshift cluster take about twice as long. The unit tests are pretty quick. I think this has something to do with the dynamic parameterization of the fixtures, as the slow down came once I introduced that logic.

File: sqlalchemy_redshift/dialect.py (hunk 2 above)
Comment (Brooke-white) [reply sqlalchemy_redshift/dialect.py]: addressed in edfb456

**Review by jklukas — APPROVED** [review]
LGTM. Kicked off integration tests again and will merge when done.

**Comment by Brooke-white** [comment]
thanks for the quick turn around, @jklukas! When do you plan on making the next release? We'd like to get this feature out to customers as soon as we can.

**Comment by jklukas** [comment]
I just pushed 0.8.6 which includes this new connector.

**Pull request merged**